tika-core/src/main/java/org/apache/tika/detect/CompositeEncodingDetector.java

```diff
@@ -71,10 +79,15 @@ public Charset detect(InputStream input, Metadata metadata) throws IOException {
     }
 
     public List<EncodingDetector> getDetectors() {
-        return Collections.unmodifiableList(detectors);
+        List<EncodingDetector> allDetectors = new LinkedList<>();
+        if (loader != null) {
+            allDetectors.addAll(loader.loadDynamicServiceProviders(EncodingDetector.class));
+        }
+        allDetectors.addAll(detectors);
+        return allDetectors;
     }
 
-    private boolean isExcluded(Collection<Class<? extends EncodingDetector>> excludeEncodingDetectors,
+    protected boolean isExcluded(Collection<Class<? extends EncodingDetector>> excludeEncodingDetectors,
```

Comment:
the getDetector of the subclass also need isExcluded

Reply:
Why does the subclass need to filter?

Reply:
becuase the subclass loads the dynamic services - this class only knows about the static

Reply:
But in CompositeDetector the isExcluded method is private too...

Implementation of CompositeDetector is bad, because both subclasses have to implement loading the dynamic services by their own. That's against the idea of inheritance.

In addition to that the sub classes of CompositeDetector do not filter based on excluded list. So then we go the simple path and do it the same way: we don't filter the dynamic.

How many implementations with CompositeEncodingDetector as parent class do exist?

Reply:
Okay then I will remove the filtering of the dynamic

CompositeEncodingDetector has one subclass (that I can find by searching on github) - i.e. the DefaultEncodingDetector